apps/consulting/posts/openai-pseudocode.md

```diff
@@ -0,0 +1,225 @@
+│                                                                                                  │
+│ import os                                                                                        │
+│ import datetime                                                                                  │
+│ import requests                                                                                  │
+│                                                                                                  │
+│ def run(repository: str) -> list:                                                                │
+│     headers = {'Authorization': f"Bearer {os.environ['GITHUB_TOKEN']}"}                          │
+│     url = f"https://api.github.com/repos/{repository}/issues"                                    │
+│     since = datetime.datetime.now() - datetime.timedelta(days=10)                                │
+│     params = {'since': since.isoformat()}                                                        │
+│     response = requests.get(url, headers=headers, params=params)                                 │
+│     issues = response.json()                                                                     │
+│     return [(i['title'], i['number'], i['created_at']) for i in issues if i['number'] % 2 != 0]  │
+│                                                                                                  │
+╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
+What code feedback would you like to provide? (leave empty for approval):
+```
+
+Already after some quick initial feedback we can see that the LLM
+takes into account the users needs. After user provided feedback the
+automated tests are run. If those tests fail or run into exceptions
+they will be automatically passed back to the LLM for corrections.
+
+```
+pass result = str(all([_[1] % 2 == 1 for _ in run("conda/conda")]))
+github issues [
+  ('Use appropriate `defaults` value for Windows is use when subdir=win-*', 12555, '2023-03-30T17:44:55Z'), 
+  ...
+  ('Document steps to debug tests with vscode', 12459, '2023-03-07T18:20:08Z')
+]
+```
+
+## Conclusion
+
+There are several key ideas here we want to highlight about why we
+think the approaches taken in `pseudocode` are unique.
+
+ * Feedback is generated from the user and the rest is guided by
+   automated testing and exception reporting back to the
+   LLM. __Autogenerated code needs guidance__.
+
+ * Declaring interfaces for LLMs to operate within. Interface design
+   is naturally a high level task which is the key to composable code.
+
+ * Separation of code declarations and generated code. Similar to
+   `.py` and `.pyc` files.
+
+We mentioned earlier that this was an experiment of the interaction
+between automated testing, user feedback, and LLMs. We want to explore
+this more and already have additional ideas we would like to
```

Comment:
"...additional ideas, like applying linting to generated LLM code via black, ..."